component/dependencies/cloudscale-metrics-collector/component/main.jsonnet

```diff
@@ -0,0 +1,108 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local inv = kap.inventory();
+local params = inv.parameters.cloudscale_metrics_collector;
+
+local namespace = 'appuio-cloud-reporting';
+local image = 'ghcr.io/vshn/cloudscale-metrics-collector:v0.2.0';
+
+
+local labels = {
+  'app.kubernetes.io/name': 'appuio-cloud-reporting',
+  'app.kubernetes.io/managed-by': 'commodore',
+  'app.kubernetes.io/part-of': 'syn',
+};
+
+
+{
+  namespace: {
+    kind: 'Namespace',
+    apiVersion: 'v1',
+    metadata: {
+      name: namespace,
+      labels+: labels,
+    },
+  },
+
+  secret: {
+    assert params.cloudscale.token != null : 'cloudscale.token must be set.',
+    kind: 'Secret',
+    apiVersion: 'v1',
+    metadata: {
+      name: 'cloudscale-api',
+      namespace: namespace,
+      labels+: labels,
+    },
+    stringData: {
+      token: params.cloudscale.token
+    }
+  },
+
+  cronjob: {
+    kind: 'CronJob',
+    apiVersion: 'batch/v1',
+    metadata: {
+      name: 'cloudscale-metrics-collector',
+      namespace: namespace,
+      labels+: labels,
+    },
+    spec: {
+      concurrencyPolicy: 'Forbid',
+      failedJobsHistoryLimit: 768,
+      jobTemplate: {
+        spec: {
+          template: {
+            spec: {
+              restartPolicy: 'OnFailure',
+              containers: [
+                {
+                  args: [
+                    'cloudscale-metrics-collector',
+                  ],
+                  command: ['sh', '-c'],
+                  env: [
+                    {
+                      name: 'password',
+                      valueFrom: {
+                        secretKeyRef: {
+                          key: 'password',
+                          name: 'reporting-db',
+                        },
+                      },
+                    },
+                    {
+                      name: 'username',
+                      valueFrom: {
+                        secretKeyRef: {
+                          key: 'username',
+                          name: 'reporting-db',
+                        },
+                      },
+                    },
+                    {
+                      name: 'ACR_DB_URL',
+                      value: 'postgres://$(username):$(password)@reporting-db.appuio-reporting.svc:5432/reporting?sslmode=disable',
```

Comment:
How is the secret managed that provides these credentials? Is it given by the other deployment?

Reply:
Yes. This component has to be deployed into the same namespace as component-appuio-cloud-reporting is deployed, otherwise it won't work. This is necessary to give it access to the existing secrets and the billing database.